pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -1655,6 +1701,65 @@ func updateBasedOnProfile(ctx *zedmanagerContext, oldProfile string) {
 	}
 }
 
+// countRunningAppsForUUIDs returns the number of app instances (from the provided uuidMap)
+// that are in a running state (BOOTING, RUNNING, HALTING, PURGING, or RESTARTING).
+func countRunningAppsForUUIDs(ctx *zedmanagerContext, uuidMap map[string]bool) (runningCount uint) {
+	sub := ctx.subAppInstanceStatus
+	items := sub.GetAll()
+	log.Noticef("countRunningAppsForUUIDs: %d items", len(items))
+	for _, s := range items {
+		status := s.(types.AppInstanceStatus)
+		log.Noticef("countRunningAppsForUUIDs: %s %s", status.UUIDandVersion, status.State)
+		// Check if this status belongs to one of the app UUIDs in our map.
+		if !uuidMap[status.UUIDandVersion.UUID.String()] {
+			continue
+		}
+		switch status.State {
+		case types.BOOTING, types.RUNNING, types.HALTING:
+			runningCount++
+		}
+	}
+	return runningCount
+}
+
+// Wait for all high-priority apps to start before launching this app.
+// Every 5 seconds, verify whether the priority apps are running.
+// If they have started, launch the app immediately; otherwise,
+// once the timeout expires, proceed to start the app regardless.
+func waitForAppsToStart(ctx *zedmanagerContext, uuids []string, config types.AppInstanceConfig) {
+	activeMap := make(map[string]bool)
```

Comment:
I don't understand what does it mean if the value of a map entry is set to `false`?

Or do you mean:
```suggestion
	activeMap := make(map[string]struct{})
```